### PR TITLE
chore: install clockface beta and fix e2e tests

### PR DESCRIPTION
**Pull request by ChitlangeSahas**
Installs [clockface beta](https://www.npmjs.com/package/@influxdata/clockface/v/3.0.0-beta.4) version. This will let us know if there are test failures.

Note for the of the UI team: Notice that this is merging into `clockface_beta_master`, that branch is a long-running branch that will take in small planned PR's to introduce clockface 3 into the UI eventually. Please come reach out to us in the `project-clockface3-integration` slack channel if you have questions and suggestions.

### CODE REVIEW

**Comment by ChitlangeSahas** [comment]
Just tagging you @daviddkkim so you can get notified about this PR-s existence. 👍

**Review by daviddkkim — APPROVED** [review]
lgtm

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 59 | :red_circle:
accessibility | 97 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=2502)

File: cypress/e2e/shared/nav.test.ts [hunk 1]
```diff
@@ -158,6 +163,14 @@ describe('navigation', () => {
...
       'labels',
     ].forEach(x => {
+      if (x === 'history') {
+        cy.getByTestID('nav-item-alerting').click()
+      } else if (x === 'variables' || x === 'templates' || x === 'labels') {
+        cy.getByTestID('nav-item-settings').click()
+      } else {
+        cy.getByTestID('nav-item-load-data').click()
+      }
+
```
Comment (ChitlangeSahas) [inline cypress/e2e/shared/nav.test.ts:173]: Because the way new navigation works, we have to open up tabs before we can click on the sub menu items. Before clockface 3, the submenu items were open by default.

File: cypress/e2e/shared/nav.test.ts [hunk 2]
```diff
@@ -9,31 +9,36 @@ describe('navigation', () => {
...
     cy.getByTestID('empty-dashboards-list').should('exist')
 
     // Tasks Index Page
-    cy.getByTestID('nav-item-tasks').click()
+    cy.getByTestID('nav-item-tasks').click({force: true})
     cy.getByTestID('tasks-page--header').should('exist')
 
     // Alerts Page
-    cy.getByTestID('nav-item-alerting').click()
+    cy.getByTestID('nav-item-alerting').click({force: true})
```
Comment (ChitlangeSahas) [inline cypress/e2e/shared/nav.test.ts:33]: These `force: true` clicks are necessary because new design overlays an `<a>` tag element over our buttons and this is the best way for now.

File: cypress/e2e/shared/flows.test.ts [hunk 3]
```diff
@@ -369,6 +369,6 @@ describe('Flows', () => {
     cy.getByTestID('resource-editable-name').click()
 
     // visualizations should exist
-    cy.getByTestID('giraffe-inner-plot').should('be.visible')
+    cy.getByTestID('giraffe-inner-plot').should('exist')
```
Comment (ChitlangeSahas) [inline cypress/e2e/shared/flows.test.ts:372]: They should exist.

File: cypress/e2e/shared/variables.test.ts [hunk 4]
```diff
@@ -343,7 +343,9 @@ describe('Variables', () => {
     cy.getByTestID('inline-labels--popover--contents').type(labelName)
     cy.getByTestID('inline-labels--create-new').click()
 
-    cy.getByTestID('create-label-form--submit').should('be.visible')
+    cy.getByTestID('create-label-form--submit')
+      .scrollIntoView()
```
Comment (ChitlangeSahas) [inline cypress/e2e/shared/variables.test.ts:347]: New spacings in the UI means we need to scroll to it.

File: cypress/e2e/shared/variables.test.ts [hunk 5]
```diff
@@ -18,7 +18,7 @@ describe('Variables', () => {
   it('can CRUD a CSV, upload, map, and query variable and search for variables based on names', () => {
     // Navigate away from and back to variables index using the nav bar
     cy.getByTestID('nav-item-dashboards').click()
-    cy.getByTestID('nav-item-settings').click()
+    cy.getByTestID('nav-item-settings').click({force: true})
```
Comment (daviddkkim) [inline cypress/e2e/shared/variables.test.ts:21]: New navigation when it is collapsed have an overlaying button for the submenu. Force true was added to go around that.

File: src/variables/components/TypeAheadVariableDropdown.tsx [hunk 6]
```diff
@@ -284,6 +284,7 @@ class TypeAheadVariableDropdown extends PureComponent<Props, MyState> {
         testID={this.props.testID || `variable-dropdown--${name}`}
         onClickAway={this.onClickAwayHere}
         menuOpen={menuOpen}
+        customFocus
```
Comment (ChitlangeSahas) [inline src/variables/components/TypeAheadVariableDropdown.tsx:287]: Should we rename this to `disableAutoFocus` or something, @daviddkkim?

File: src/shared/components/DurationInput.tsx [hunk 7]
```diff
@@ -92,6 +92,11 @@ const DurationInput: FC<Props> = ({
           status={inputStatus}
           onChange={e => onChange(e.target.value)}
           onFocus={() => setIsFocused(true)}
+          onKeyPress={e => {
```
Comment (ChitlangeSahas) [inline src/shared/components/DurationInput.tsx:95]: Note to us, rename this to `evt` or `event`.

File: src/variables/components/TypeAheadVariableDropdown.tsx (hunk 6 above)
Comment (daviddkkim) [reply src/variables/components/TypeAheadVariableDropdown.tsx:287]: yeah I think that'd be the better name. Let's get around to that after we finish fixing the tests

File: cypress/e2e/shared/nav.test.ts [hunk 8]
```diff
@@ -157,11 +162,23 @@ describe('navigation', () => {
...
+        cy.getByTestID('nav-item-alerting').click()
+      } else if (
+        navItem === 'variables' ||
+        navItem === 'templates' ||
+        navItem === 'labels'
+      ) {
+        cy.getByTestID('nav-item-settings').click()
+      } else {
+        cy.getByTestID('nav-item-load-data').click()
+      }
```
Comment (hoorayimhelping) [inline cypress/e2e/shared/nav.test.ts:176]: why is this needed now?

**Review by hoorayimhelping — COMMENTED** [review]
nice annotations on the PR by both you @ChitlangeSahas and @daviddkkim. just a couple of questions.

File: cypress/e2e/shared/dashboardsRefresh.test.ts [hunk 9]
```diff
@@ -155,14 +157,15 @@ describe('Dashboard refresh', () => {
       cy.wait('@refreshQuery')
       cy.wait(5000)
       cy.getByTestID('enable-auto-refresh-button').then(el => {
-        expect(el[0].innerText).to.equal('Enable Auto Refresh')
+        expect(el[0].innerText.toLowerCase()).to.equal('enable auto refresh')
```
[the remark this answers is not in the document]
Comment (ChitlangeSahas) [reply cypress/e2e/shared/dashboardsRefresh.test.ts:160]: The new buttons have all caps in the text, like `ENABLE AUTO REFRESH`, I think we should test for the all caps version, @daviddkkim ?

```suggestion
        expect(el[0].innerText).to.equal('ENABLE AUTO REFRESH')
```

File: cypress/e2e/shared/explorer.test.ts [hunk 10]
```diff
@@ -66,6 +66,7 @@ describe('DataExplorer', () => {
       cy.get('.cf-tree-nav--toggle').click()
       // Can't use the testID to select this nav item because Clockface is silly and uses the same testID twice
       // Issue: https://github.com/influxdata/clockface/issues/539
+      cy.getByTestID('nav-item-load-data').click()
       cy.get('.cf-tree-nav--sub-item-label')
```
Comment (genehynson) [inline cypress/e2e/shared/explorer.test.ts:70]: this comment is supposed to be on the line underneath this change

File: cypress/e2e/oss/onboarding.test.ts [hunk 11]
```diff
@@ -182,17 +182,19 @@ describe('Onboarding', () => {
 
     cy.getByTestID('next')
       .should('be.disabled')
-      .children('.cf-button--label')
-      .contains('Continue')
+      .then(el => {
+        expect(el[0].innerText.toLowerCase()).to.equal('continue')
+      })
```
[the remark this answers is not in the document]
Comment (genehynson) [reply cypress/e2e/oss/onboarding.test.ts:187]: +1 to this, we should use `.contains`

File: cypress/e2e/shared/dashboardsRefresh.test.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (daviddkkim) [reply cypress/e2e/shared/dashboardsRefresh.test.ts:160]: yeah new buttons have all caps. I wanted to make the test less fragile by not being case dependent but if we want to test casing as well, Sahas's suggestion looks good

File: cypress/e2e/shared/flows.test.ts (hunk 3 above)
Comment (genehynson) [reply cypress/e2e/shared/flows.test.ts:372]: ...but not be visible?

File: cypress/e2e/shared/nav.test.ts (hunk 8 above)
Comment (daviddkkim) [reply cypress/e2e/shared/nav.test.ts:176]: In the current Nav, all sub menu are expanded by default. With the new nav, submenus are hidden by default, visible only when expanded. So this logic takes care of expanding the submenu

File: cypress/e2e/shared/explorerVisualizations.test.ts [hunk 12]
```diff
@@ -415,6 +415,8 @@ describe('visualizations', () => {
     })
 
     it('can view table data & scroll to bottom', () => {
+      // to fit all the elements in screen we force a view port size
+      cy.viewport(1200, 1000)
```
Comment (genehynson) [inline cypress/e2e/shared/explorerVisualizations.test.ts:419]: just a word of caution, we had issues in the past when tests were modifying the viewport in the middle of the test. Would be preferred if we could avoid this with another solution

File: cypress/e2e/shared/nav.test.ts (hunk 8 above)
Comment (ChitlangeSahas) [reply cypress/e2e/shared/nav.test.ts:176]: In our previous UI, sub menu items were open and visible by default like this: 

Screen Shot 2021-09-03 at 11 24 55 AM


new UI, we have to click on them to open the sub menu items. So, this adds a little check.

Screen Shot 2021-09-03 at 11 28 14 AM

File: cypress/e2e/shared/simpleTable.test.ts [hunk 13]
```diff
@@ -6,6 +6,8 @@ describe('simple table interactions', () => {
   const simpleLarge = 'simple-large'
   const simpleOverflow = 'simple-overflow'
   beforeEach(() => {
+    // to fit all the elements in screen we force a view port size
+    cy.viewport(1200, 1000)
```
Comment (genehynson) [inline cypress/e2e/shared/simpleTable.test.ts:10]: same as other viewport comment above :)

**Review by genehynson — COMMENTED** [review]
added some comments around test changes

File: cypress/e2e/shared/explorerVisualizations.test.ts (hunk 12 above)
Comment (hoorayimhelping) [reply cypress/e2e/shared/explorerVisualizations.test.ts:419]: @genehynson is it possible to set the viewport globally? it seems to be pretty small for our app

File: cypress/e2e/shared/nav.test.ts (hunk 2 above)
Comment (genehynson) [reply cypress/e2e/shared/nav.test.ts:33]: I'm wondering if maybe we should create a Cypress command for clicking these nav items? Especially if this is something that we're going to have to do in every test that wants to click a nav bar element. Seems like something that people are going to be confused about as they are adding new tests

File: cypress/e2e/shared/explorer.test.ts (hunk 10 above)
Comment (ChitlangeSahas) [reply cypress/e2e/shared/explorer.test.ts:70]: So true, i'll fix that 👍

File: cypress/e2e/shared/nav.test.ts (hunk 2 above)
Comment (genehynson) [reply cypress/e2e/shared/nav.test.ts:33]: Like I'm thinking a Cypress command that expands the nav bar and clicks the element (instead of doing `force: true`. The signature would be like `cy.clickNavBar('nav-item-tasks')`

Comment (ChitlangeSahas) [reply cypress/e2e/shared/nav.test.ts:33]: I think this is a solid idea, it'll make more sense yeah!

Comment (ChitlangeSahas) [reply cypress/e2e/shared/nav.test.ts:33]: I'll try to do this by seeing how similar functions were done before in the cypress setup.

File: cypress/e2e/shared/explorerVisualizations.test.ts (hunk 12 above)
Comment (genehynson) [reply cypress/e2e/shared/explorerVisualizations.test.ts:419]: totes! https://docs.cypress.io/api/commands/viewport#Defaults Just toss it into the `cypress.json` file in the UI repo

File: cypress/e2e/shared/nav.test.ts (hunk 2 above)
Comment (genehynson) [reply cypress/e2e/shared/nav.test.ts:33]: 💯  awesome! yeah check the `commands.ts` file!

File: cypress/e2e/oss/onboarding.test.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (daviddkkim) [reply cypress/e2e/oss/onboarding.test.ts:187]: I got this pattern from an existing test for auto-refresh.

I'll try with contains 'continue'

[the remark this answers is not in the document]
Comment (genehynson) [reply cypress/e2e/oss/onboarding.test.ts:187]: hmm yeah I see those references. Should probably refactor those too. Not sure why .contains wouldn't suffice

[the remark this answers is not in the document]
Comment (daviddkkim) [reply cypress/e2e/oss/onboarding.test.ts]: Screen Shot 2021-09-03 at 4 07 40 PM

[the remark this answers is not in the document]
Comment (daviddkkim) [reply cypress/e2e/oss/onboarding.test.ts]: el.innertext assert works but not contains('CONTINUE') :/

File: cypress/e2e/shared/dashboardsRefresh.test.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (genehynson) [reply cypress/e2e/shared/dashboardsRefresh.test.ts:160]: btw Cypress has a "matchCase" option: https://docs.cypress.io/api/commands/contains#Case-Sensitivity

[the remark this answers is not in the document]
Comment (daviddkkim) [reply cypress/e2e/shared/dashboardsRefresh.test.ts:160]: you're a saint

File: cypress/e2e/oss/onboarding.test.ts [hunk 14]
```diff
@@ -213,10 +209,9 @@ describe('Onboarding', () => {
     // check password too short
     cy.getByTestID('form--element-error').should('contain.text', '8')
 
-    cy.getByTestID('next')
-      .should('be.disabled')
-      .children('.cf-button--label')
-      .contains('Continue')
+    cy.getByTestID('next').should('be.disabled')
+
+    cy.contains('CONTINUE')
```
Comment (genehynson) [inline cypress/e2e/oss/onboarding.test.ts:214]: this is a smell to me - what element are we checking contains "CONTINUE"?

File: cypress/e2e/oss/onboarding.test.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (genehynson) [reply cypress/e2e/oss/onboarding.test.ts]: I don't think `cy.contains('CONTINUE')` is the solution. You would want to `cy.getByTestID(...).contains('CONTINUE')`

[the remark this answers is not in the document]
Comment (daviddkkim) [reply cypress/e2e/oss/onboarding.test.ts]: ^ that's what was broken in the test in the first place though. Trying a few different things brb

[the remark this answers is not in the document]
Comment (daviddkkim) [reply cypress/e2e/oss/onboarding.test.ts]: @genehynson @hoorayimhelping It makes no sense but I think `adding{ matchCase: false}` makes the test pass... with the original logic `cy.getByTestID(...).contains('CONTINUE')  `

[the remark this answers is not in the document]
Comment (genehynson) [reply cypress/e2e/oss/onboarding.test.ts]: nice!! that's pretty bizarre but works for me 😄

File: cypress/e2e/oss/onboarding.test.ts (hunk 14 above)
Comment (daviddkkim) [reply cypress/e2e/oss/onboarding.test.ts]: It's a button in the onboarding form for OSS; That button has the text content 'CONTINUE'. I am not sure why the original test writer was checking that button's text content like million times in the test. 

I refactored this test to be more coherent by using .contains on the actual button

Comment (genehynson) [reply cypress/e2e/oss/onboarding.test.ts]: perfect!

File: cypress/e2e/oss/onboarding.test.ts [hunk 15]
```diff
@@ -71,16 +71,12 @@ describe('Onboarding', () => {
 
     cy.getByTestID('next').should('be.disabled')
 
-    cy.getByTestID('next')
-      .children('.cf-button--label')
-      .contains('Continue')
+    cy.getByTestID('next').contains('CONTINUE', {matchCase: false})
```
Comment (daviddkkim) [inline cypress/e2e/oss/onboarding.test.ts:74]: for some reason, adding matchCase:false passes the test

File: cypress/e2e/shared/nav.test.ts [hunk 16]
```diff
@@ -9,31 +9,36 @@ describe('navigation', () => {
...
+    cy.getByTestID('tabs').within(() => {
+      // buckets tab
+      cy.get('[id="buckets"]').click()
+      cy.url().should('contain', 'buckets')
+    })
     cy.getByTestID('load-data--header').should('exist')
 
     // Data Explorer Page
-    cy.getByTestID('nav-item-data-explorer').click()
+    cy.clickNavBarItem('nav-item-data-explorer')
```
Comment (daviddkkim) [inline cypress/e2e/shared/nav.test.ts:21]: @genehynson custom navbar click command

Comment (genehynson) [reply cypress/e2e/shared/nav.test.ts:21]: love it!

File: cypress/e2e/shared/flows.test.ts (hunk 3 above)
Comment (genehynson) [reply cypress/e2e/shared/flows.test.ts:372]: I'm wondering why we need to make this change. Did the visualization just get pushed further down the screen so we need to scrollIntoView?

File: cypress/support/commands.ts [hunk 17]
```diff
@@ -297,6 +297,10 @@ export const upsertSecret = (
   })
 }
 
+export const clickNavBarItem = (testID: string): Cypress.Chainable => {
+  return cy.getByTestID(testID).click({force: true})
+}
```
Comment (genehynson) [inline cypress/support/commands.ts:302]: this is ok for now - but I was hoping this function would actually do the work to expand the nav bar menu and then select the menu item

Comment (ChitlangeSahas) [reply cypress/support/commands.ts:302]: Ah! I see what you mean. I think me and David thought the same thing that we just want to wrap it up into a function.

Comment (genehynson) [reply cypress/support/commands.ts:302]: The reason I think it would be a good idea to actually expand the menu and then click the visible nav item is because that's how the users will do it real life. But it's not a big deal - now that you've put it in a function you could always make that improvement later! Now it only has to be changed in one spot 😄

Comment (genehynson) [reply cypress/support/commands.ts:302]: Maybe put a "//TODO" comment in the function for now or something

File: cypress/e2e/shared/simpleTable.test.ts (hunk 13 above)
Comment (ChitlangeSahas) [reply cypress/e2e/shared/simpleTable.test.ts]: Fixed 👍

File: cypress/e2e/shared/flows.test.ts (hunk 3 above)
Comment (ChitlangeSahas) [reply cypress/e2e/shared/flows.test.ts]: Just used the scroll into view and it worked! not sure why it didn't previously!

**Review by genehynson — APPROVED** [review]

File: cypress/e2e/shared/explorerVisualizations.test.ts (hunk 12 above)
Comment (ChitlangeSahas) [reply cypress/e2e/shared/explorerVisualizations.test.ts]: ✅

File: cypress/support/commands.ts [hunk 18]
```diff
@@ -297,6 +297,11 @@ export const upsertSecret = (
   })
 }
 
+// TODO: (sahas) need to make this function to actually mimic the navbar functionality
```
Comment (hoorayimhelping) [inline cypress/support/commands.ts:300]: can you write up a quick ticket for this todo please?

**Review by hoorayimhelping — APPROVED** [review]

File: cypress/support/commands.ts (hunk 18 above)
Comment (ChitlangeSahas) [reply cypress/support/commands.ts:300]: Just did 👍  

https://github.com/influxdata/ui/issues/2548
